src/compiler/checker.ts

```diff
@@ -4504,18 +4526,34 @@ namespace ts {
                         parameters.push(paramSymbol);
                     }
 
-                    if (param.type && param.type.kind === SyntaxKind.StringLiteralType) {
-                        hasStringLiterals = true;
-                    }
+                    if (isJSDocParameterTag(param)) {

```

Comment:
You already know whether or not you've originated from JSDoc-land or TS-land, so consider caching the value once and branching on that.